public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -344,3 +360,59 @@ export function extractLabelMatchers(tokens: Array<string | Token>): AbstractLab
 
   return labelMatchers;
 }
+
+/**
+ * Calculates new interval "snapped" to the closest Nth minute, depending on cache level datasource setting
+ * @param cacheLevel
+ * @param range
+ */
+export function getRangeSnapInterval(
+  cacheLevel: PrometheusCacheLevel,
+  range: TimeRange
+): { start: string; end: string } {
+  // Don't round the range if we're not caching
+  if (cacheLevel === PrometheusCacheLevel.None) {
+    return {
+      start: getPrometheusTime(range.from, false).toString(),
+      end: getPrometheusTime(range.to, true).toString(),
```

Comment:
Is rounding up intentional here?

Reply:
Yup, should be the same behavior as current